airbyte-workers/src/test/java/io/airbyte/commons/temporal/scheduling/RouterServiceTest.java

```diff
@@ -40,21 +40,21 @@ class RouterServiceTest {
   void init() {
     routerService = new RouterService(mConfigRepository, mTaskQueueMapper);
 
-    Mockito.when(mTaskQueueMapper.getTaskQueue(Geography.AUTO)).thenReturn(US_TASK_QUEUE);
-    Mockito.when(mTaskQueueMapper.getTaskQueue(Geography.US)).thenReturn(US_TASK_QUEUE);
-    Mockito.when(mTaskQueueMapper.getTaskQueue(Geography.EU)).thenReturn(EU_TASK_QUEUE);
+    Mockito.when(mTaskQueueMapper.getTaskQueue(Geography.AUTO, TemporalJobType.SYNC)).thenReturn(US_TASK_QUEUE);
+    Mockito.when(mTaskQueueMapper.getTaskQueue(Geography.US, TemporalJobType.SYNC)).thenReturn(US_TASK_QUEUE);
+    Mockito.when(mTaskQueueMapper.getTaskQueue(Geography.EU, TemporalJobType.SYNC)).thenReturn(EU_TASK_QUEUE);
   }
 
   @Test
```

Comment:
You could turn this into a @ParameterizedTest where it takes in an @EnumSource for `TemporalJobType` and verifies that it works correctly for everything, instead of just SYNC. Here's an example test that uses an EnumSource: https://github.com/airbytehq/airbyte-platform-internal/blob/42ac3ae31fb7a3ce16862597c637ff63e8c307f1/cloud-config/cloud-config-persistence/src/test/java/io/airbyte/cloud/config/persistence/billing/BillingPersistenceTest.java#L201-L225

Reply:
I guess problem is we are not doing routing logic for all TemporalJobType enums.

I think testing sync only at RouterServiceTest is fine - the other tests can be tested within taskQueueMapper logic.

Reply:
Oh, the @EnumSource takes in a list of the values you want to test: https://www.baeldung.com/parameterized-tests-junit-5#3-enum

Reply:
But yeah, not critical to cover everything here, I'm mostly just trying to spread awareness of ParameterizedTest because of how useful it is!